libkbfs/prefetcher.go

```diff
@@ -122,12 +122,6 @@ func (p *blockPrefetcher) run() {
 func (p *blockPrefetcher) request(priority int, kmd KeyMetadata,
 	ptr BlockPointer, block Block, entryName string,
 	doneCh, errCh chan<- struct{}) error {
-	if _, err := p.config.BlockCache().Get(ptr); err == nil {
-		return nil
-	}
```

Comment:
This was a major PitA to find: it was causing some huge goroutine leaks and unfinished prefetches.

Reply:
Do we have anything calling `checkDataVersion` in the prefetcher path anymore?

Reply:
We do not. What do you suggest would be the right approach here? I tried adding it in `BlockRetrievalQueue` after `checkCaches`, but it broke a ton of tests.

Reply:
Hmm, I'm not actually sure why we were checking the version that came out of the cache.  We should probably maintain an invariant that we only put useful blocks in the cache, ones that we can read.

Ideally it would be checked any time a block is fetched from a server, before it's put into the cache.

Reply:
@strib: we weren't checking the version that came out of the cache, we were only checking if it _didn't_ come from the cache.

I tried to put that into the block retriever instead, but then `block_ops` tests broke because they were expecting different errors. It would probably be a pretty mechanical fix to update the `DataVersion` fields on those tests so the correct error is returned.

That said, `checkDataVersion` only checks pointers + paths, not blocks themselves. So we'd be checking _before_ retrieving from the server, not after.

Reply:
Oh right, good point, supposed to check them before fetching.  I think it's a good guard to have, but up to you if you want it in this PR.

Reply:
I'll add it and fix the tests.